### PR TITLE
docs(notebooks): add partial folding guide

### PR DESCRIPTION
**Pull request by ritajitmajumdar1**
### Summary

This pull request is for the how-to guide on partial folding with ZNE which now works with Qiskit 1.0.

### Details and comments

Requesting feedback

### CODE REVIEW

**Comment by pedrorrivero** [comment]
Hi @ritajitmajumdar1 ,

Please add the `wire-cutting` guide in a separate branch and open a new PR.

**Review by miamico — CHANGES_REQUESTED** [review]
The guide looks great! Some minor comments:

- This function `build_compute_uncompute_qaoa_circuit` has a PR to be hosted in the repo, let's keep it like this for now but remember to import it once everything is merged.

- The noise factors used in the regular zne comparison `zne_options.resilience.noise_factors = [1, 3, 5]` are too large (expectation values are 3 are already 0), can we try [1,2,3] instead?

**Review by ashsaki — CHANGES_REQUESTED** [review]

**Comment by ritajitmajumdar1** [comment]
@ashsaki I have just updated the notebook as per your reviews. @miamico your reviews on using noise factors [1,2,3] are also incorporated. Although, I may be wrong, isn't [1,3,5] the default values in Primitives?

**Comment by ritajitmajumdar1** [comment]
@ashsaki @pedrorrivero @miamico, I was testing out the ZNE multiple times. Here is the code snippet in case I am doing something wrong:

```
estimator = Estimator(session=session)
estimator.options.resilience_level=2 # applies ZNE
estimator.options.resilience.zne.noise_factors=[1,1.2,1.4]
estimator.options.resilience.zne.extrapolator='linear'
estimator.options.default_shots=5000

job = estimator.run([(trans_qaoa, trans_obs)])
```

I am attaching the figure after running this code with the same trans_qaoa and trans_obs for 20 trials, and see the variance in the extrapolated result. Is such a variance acceptable?
test

**Comment by miamico** [comment]
I'm not surprised to see somewhat high variance in the extrapolated values though. This will depend on the fit on the measured expectation values at the various noise factor and if those are a bit noisy, the fit will end up being noisy as well. If it was possible to increment the number of circuits used to get the approximate partial increase in noise factor, I'd expect to see more stable extrapolations.

**Review by miamico — DISMISSED** [review]

**Comment by ashsaki** [comment]
@ritajitmajumdar1 let me know if your done testing on the HW and make sure to check if the returned `zne_noise_factors` is `0.0`. Once you confirm these, I will approve the PR.

File: docs/notebooks/partial_folding.ipynb [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
...Kjg8mN/FxoaqvT09Bz39erVS02aNFFsbKxKly6tBg0aXHVspUqVlJKSooyMDAUHB8vpdOrAgQOqXLlyrq+Xnp6uIkWKuLiWAJA7X6iD7qAOAvAUX6qDEydO1PLly7Vu3TqFhobmuSx1EIAn+FINzPLAAw9o4MCBOnHihEqVKnXVZaiBADzFF+pgQkKCkpKSVL16dUnSkSNHFB8fr5SUFA0aNOiqY6iDAAAAgcXM7dKNGzd6LGde32s7HA5J0tixY3OMMXO7WGLbGAACAX3wSq7sJ5fogwAQKOiFlxk5bkaiFwJAIKAPuo8+CACBgV7o3rm1Er0QAAAAAAAAAAAAAACYJ8jbAQDAVWfOnFGXLl303HPP6bbbbnP5saupV6+edu7cmeO+ihUrqmbNmho4cKCGDBmS69iyZcvqlltu0TvvvCNJWrZsmSIjI3XDDTfkOmbHjh2Kjo7ONxcA5MVX6qA7qIMAPMGX6uCkSZO0cOFCrV27ViVLlsz39aiDAK6Vr9TA1NRUHT58OPv2ihUrVKpUKYWHh+c6hhoIwBN8pQ4OGjRIKSkpSkpKUlJSkpo0aaI33ngjzwvJUgcBAAACh9nbpZ7kzvfaZm4XS2wbA4C/ow/+yZ395BJ9EAACAb3wMqPHzUj0QgDwd/TBa0MfBAD/Ry90/7kleiEAAAAAAAAAAAAAADAPkywA8BvTp0/Xzp07NWfOHMXExOT4mTZtWq6P/fWk1ixdu3bVp59+esX9vXv3VkZGhu677748s8yePVuzZ89WjRo19PLLL2v+/Pm5LpuUlKRLly5xEBiAa+ZLdXDAgAGKjIxUcnKyWrVqlefBsNRBAJ7iK3UwOTlZw4cPV2pqquLi4hQTE6PGjRvnujx1EIAn+EoNPHXqlDp37qy6desqOjpar7/+ulatWiWHw3HV5amBADzFV+qgUdRBwPcFBQWpYcOGCgria1MAQP6s2C7NT17f0fTv318rV67Mvm3ke+0sZmwXS2wbA0AgoA/+yeh+cok+CACBgl74J6PHzUj0QgAIBPRB13JcDX0QAAIDvfAyo89NL/RdHDcFAAAAAAAAAAAAAAgEwd4OAACueuKJJ/TEE0/k+vhTTz3l8nP17dtXsbGxGjt2rMLCwrLvr1y5sipWrKhChQrlOf7GG2/Uxo0bXXqtWbNmaeTIkXmeSAsArvClOjh79myXX4s6C
```
Comment (ashsaki) [inline docs/notebooks/partial_folding.ipynb:331]: Simplify to
```suggestion
    "trans_obs = observable.apply_layout(trans_qaoa.layout)"
```

**Review by ashsaki — CHANGES_REQUESTED** [review]

**Review by ashsaki — DISMISSED** [review]

**Review by pedrorrivero — CHANGES_REQUESTED** [review]
Thanks so much @ritajitmajumdar1 ! Great work overall

A few comments to address:

- [x] _Noise amplification_ (i.e. not ZNE) is categorized in the following types:
  - Analogue → Hardware based (e.g. pulse)
  - Digital → Circuit based
    - Foldings
    - PEA
- [x] Add links to references
- [x] Adjust your notation to match that in the scaling notebook(s) (e.g. $U(U^\dagger U)^f$, $\lambda = 2f + 1$)
- [x] Make sure that you include all contents from the scaling notebook(s) here.
- [x] Update the circuit building functions to latest
- [x] Why $p=5$ but two-qubit depth is 40? It should be 20 after the compute-uncompute. I believe this is related to transpilation of parametrized circuits.
- [x] I think it would be beneficial to expand a bit on the theory of partial foldings, maybe point to the dZNE best practices paper
- [x] Deactivate measurement and gate twirling, as well as TREX from resilience level 2
- [ ] Add a section title for the references

**Comment by ritajitmajumdar1** [comment]
Made the changes. I have added links to related paper within the text, so removed reference section.

**Review by pedrorrivero — APPROVED** [review]
Thanks @ritajitmajumdar1, good job! LGTM 😊

**Pull request merged**